pkg/gce-pd-csi-driver/node.go

```diff
@@ -264,44 +273,45 @@ func (ns *GCENodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStage
 	if part, ok := req.GetVolumeContext()[common.VolumeAttributePartition]; ok {
 		partition = part
 	}
+	devicePath, err := GetDevicePath(ns, volumeID, partition)
 
-	devicePath, err := ns.getDevicePath(volumeID, partition)
 	if err != nil {
 		return nil, status.Error(codes.Internal, fmt.Sprintf("Error when getting device path: %v", err))
 	}
 
 	klog.V(4).Infof("Successfully found attached GCE PD %q at device path %s.", volumeKey.Name, devicePath)
 
-	// Part 2: Check if mount already exists at targetpath
-	notMnt, err := ns.Mounter.Interface.IsLikelyNotMountPoint(stagingTargetPath)
-	if err != nil {
-		if os.IsNotExist(err) {
-			if err := os.MkdirAll(stagingTargetPath, 0750); err != nil {
-				return nil, status.Error(codes.Internal, fmt.Sprintf("Failed to create directory (%q): %v", stagingTargetPath, err))
+	if runtime.GOOS != "windows" {
```

Comment:
what happens on windows if we call nodestage twice? Will the 2nd time be a noop?

Reply:
similar question for nodepublish

Reply:
For nodestagevolume, it will call csi-proxy to check whether disk is already initialized, partitioned, and formatted. If all is done,  it will return success. 

For nodepublishvolume, if it is already mounted, yes, it will return immediately.

Reply:
for node publish it seems like we will always unmount if it already exists, so we would never hit that path?

Reply:
https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/483/files#diff-37c748634a23b1ed91abeaa8a6d21525R108
if it is mounted already, it will return without doing anything, right?

Reply:
Oh I see now, thanks!